File: CH8/StartWithJEE8/StartWithJEE8-ejb/src/main/java/com/packt/delivery/abstraction/entity/Token.java

package com.packt.delivery.abstraction.entity;

import java.util.Objects;

public class Token {
    private String userName;
    private String userEmail;
    private String accessToken;
    private String refreshToken;
    private String expiresIn;

    public Token() {
    }

    
    public Token(String userName, String userEmail,String accessToken, String refreshToken, String expiresIn) {
        this.userName = userName;
        this.userEmail = userEmail;
        this.accessToken = accessToken;
        this.refreshToken = refreshToken;
        this.expiresIn = expiresIn;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getUserEmail() {
        return userEmail;
    }

    public String getAccessToken() {
        return accessToken;
    }

    public String getRefreshToken() {
        return refreshToken;
    }

    public String getExpiresIn() {
        return expiresIn;
    }

    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }

    public void setRefreshToken(String refreshToken) {
        this.refreshToken = refreshToken;
    }

    public void setExpiresIn(String expiresIn) {
        this.expiresIn = expiresIn;
    }

    public void setUserEmail(String userEmail) {
        this.userEmail = userEmail;
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 23 * hash + Objects.hashCode(this.userName);
        hash = 23 * hash + Objects.hashCode(this.userEmail);
        hash = 23 * hash + Objects.hashCode(this.accessToken);
        hash = 23 * hash + Objects.hashCode(this.refreshToken);
        hash = 23 * hash + Objects.hashCode(this.expiresIn);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Token other = (Token) obj;
        if (!Objects.equals(this.userName, other.userName)) {
            return false;
        }
        if (!Objects.equals(this.userEmail, other.userEmail)) {
            return false;
        }
        if (!Objects.equals(this.accessToken, other.accessToken)) {
            return false;
        }
        if (!Objects.equals(this.refreshToken, other.refreshToken)) {
            return false;
        }
        if (!Objects.equals(this.expiresIn, other.expiresIn)) {
            return false;
        }
        return true;
    }
    
    
    
}

File: CH8/startwithvuejs/src/entities/Token.ts

export class Token {
    userName: string = ''
    userEmail: string = ''
    accessToken: string = ''
    refreshToken: string = ''
    expiresOn: string = ''

    static emptyToken () {
      let delivery:Token = new Token()

      return delivery
    }

    static newToken (userName: string, userEmail: string, accessToken: string, refreshToken: string, expiresOn: string) {
      let token:Token = new Token()

      token.userName = userName
      token.userEmail = userEmail
      token.accessToken = accessToken
      token.refreshToken = refreshToken
      token.expiresOn = expiresOn

      return token
    }

    copyToken (tokenToCopy: Token) {
      this.userName = tokenToCopy.userName
      this.userEmail = tokenToCopy.userEmail
      this.accessToken = tokenToCopy.accessToken
      this.refreshToken = tokenToCopy.refreshToken
      this.expiresOn = tokenToCopy.expiresOn
    }
}
